### PR TITLE
Upgrade guide/5.7/cin 05446

### PR DESCRIPTION
**Pull request by colinnashcinchy**
Fixes # 05446

## Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Doc update

# Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] No testing. Information is from Google Doc.

**Test configuration**:
* Doc version: 5.7

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have done an initial review
- [ ] I have updated any internal documentation
- [ ] My changes generate no new warnings
- [ ] (Optional) I have validated any automated changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have applied any changes downstream and there is no impact on related branches

# Notes

@jadhavnitind I have some comments on possible missing items. I'll comment them for you to review and approve and/or make suggestions.

### CODE REVIEW

File: deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+description: >
+This page details the instructions for upgrading your Cinchy platform to v5.7
+on Kubernetes
+
+---
+
+# v5.7 (Kubernetes)
+
+## What's new
```
Comment (colinnashcinchy) [inline deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md:11]: This was missing from previous upgrade guides. I think it's helpful to know the specific changes.

File: deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+   ```bash
+   dotnet Cinchy.DevOps.Automations.dll "deployment.json"
+   ```
+1. Commit all of your changes (if there were any) in each repository.
+1. If there were any changes in your **cinchy.argocd** repository you may need to redeploy ArgoCD.Launch a shell/terminal with the working directory set to the root of the **cinchy.argocd** repository.
+1. Execute the following command to deploy ArgoCD:
+   ```bash
+   bash deploy_argocd.sh
+   ```
+1. Execute the following command from the **cinchy.argocd** repository:
```
Comment (colinnashcinchy) [inline deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md]: This is verbatim from the email sent @jadhavnitind. I'm not sure if there's a missing item here, as the colon seems to indicate there should be code below. Not sure if it's a typo.

File: deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+1. Execute the following command to deploy ArgoCD:
+
+   ```bash
+   bash deploy_cluster_components.sh
+   bash deploy_cinchy_components.sh
+   ```
+
+1. Log in to your ArgoCD application console and refresh the apps to confirm all changes.
+
+### Upgrade AWS EKS and AKS
```
Comment (colinnashcinchy) [inline deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md]: This section below needs a good review. I broke it up into smaller pieces to make it easier to digest. Please let me know if it still reads as intended.

File: deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    ```
+
+### Deploy
+
+1. Change the AKS version in DevOps automation tools `deployment.json `against `kubernetes_version` and `orchestrator_version` key values. From a shell/terminal, go to the **cinchy.devops.automations** directory location and execute the following command:
+    ```bash
+    dotnet Cinchy.DevOps.Automations.dll "deployment.json"
+    ```
+1. In the **Terraform > Azure** directory, a new folder named `aks_cluster` should be updated with the AKS version. In that directory is a subdirectory with the same name as the newly updated cluster.
+1. Launch a shell/terminal with the working directory set to the cluster directory within the cinchy.terraform repo and run `az login`.
```
Comment (colinnashcinchy) [inline deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md]: This was changed as `az login` was on a different line with no reference to running it. However earlier in the doc `az login` was mentioned as a command, so I adjusted it. @jadhavnitind this should be approved by you.

File: deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md (hunk 2 above)
Comment (ninjadhav) [reply deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md]: Please remove this line.

File: deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md (hunk 3 above)
Comment (ninjadhav) [reply deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md]: Looks good to me. I am going send couple of changes over email as well.

File: deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md (hunk 1 above)
Comment (ninjadhav) [reply deployment-guide/upgrade-guides/upgrading-cinchy-versions/kubernetes-upgrades/v5.7-kubernetes.md:11]: Definitely it helps.

**Comment by colinnashcinchy** [comment]
@jadhavnitind changes made as per email:

> 
> Remove the highlighted section; this is going to be taken care of at step 7. 
> Redeploy components
> After the CRDs are deleted then redeploy logging-operator and kube-prometheus-stack app with below command. Change directory to cinchy.argocd repository:
> 
> bash deploy_cluster_components.sh
> 
> 
> - Below highlighted sections need to be updated/changed.
> 
> The Azure AKS and AWS EKS version supports up to 1.27
> Upgraded ArgoCD from 2.1.7 to v2.7.6
> Remove keyword (Warning) from all the places. it was to highlight that section in red color or some kind of warning.
> 
> 
> 
> 
> 
> 
> Sync ArgoCD Apps Upgrade and Redeploy components 
> Open a shell/terminal from the cinchy.devops.automations directory and execute the following command:
> 
> dotnet Cinchy.DevOps.Automations.dll "deployment.json"
> Commit all of your changes (if there were any) in each repository.
> 
> If there were any changes in your cinchy.argocd repository you may need to redeploy ArgoCD.Launch a shell/terminal with the working directory set to the root of the cinchy.argocd repository.
> 
> Execute the following command to deploy ArgoCD:
> 
> bash deploy_argocd.sh
> Execute the following command from the cinchy.argocd repository:
> 
> Validate ArgoCD pods are running and check that ArgoCD is upgraded v2.7.6 by accessing the ArgoCD application Console. Redis will take a few minutes to recover.
> 
> Execute the following command to deploy cluster components and Cinchy components:
> 
> bash deploy_cluster_components.sh
> bash deploy_cinchy_components.sh
> You may see a couple of ArgoCD apps out of sync that are because you had deleted the logging operator. Please sync them manually. Redis will take a few minutes to recover.
> Upgrade AWS EKS and Azure AKS
> For AWS export credentials and for Azure run the az login command if required.
> Delete ArgoCD apps Delete Cluster components, Cinchy components and Argocd apps
> kubectl delete -k environment_kustomizations/CLUSTER_NAME/cluster_components
> kubectl delete -k environment_kustomizations/CLUSTER_NAME/ENVIRONMENT_NAME/cinchy
> If cluster components deletion takes longer than 10 minutes, run the command below. Check all the pods are deleted except Kubernetes default namespace(kube-system)pods.
> Verify the deletion of pods for all the namespaces with the kubectl get pods -A command. If the namespaces and pods are not deleted for some of the cluster components then delete namespace manually with command "kubectl delete ns NAMESPACE"
> 
> Please add the last line to something similar to below.
> Before accepting the change, carefully verify that it meets your expectations and ensures the protection of your database and any other resources. This command will create, update, or destroy vnet, subnet, AKS cluster, and AKS node groups. Take the time to thoroughly review the changes before proceeding.

**Review by ninjadhav — APPROVED** [review]
Looks good to me.

**Pull request merged**